Hestia/app/src/main/java/hestia/UI/SingleFragmentActivity.java

```diff
@@ -51,11 +49,16 @@ public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_fragment);
 
+        /**
         backendInteractor =  ((HestiaApplication)this.getApplication()).getBackendInteractor();
         if(backendInteractor.getIp()==null){
             showIpDialog();
         }
-
+         */
+        cache = Cache.getInstance();
+        if(cache.getIp() == null) {
```

Comment:
In general, I think methods should never return null, as it contains no information. We should at some point change the getIp() method.